src/test/java/nl/tudelft/ewi/devhub/server/backend/warnings/LargeCommitWarningGeneratorTest.java

```diff
@@ -0,0 +1,137 @@
+package nl.tudelft.ewi.devhub.server.backend.warnings;
+
+import com.google.common.collect.Sets;
+import nl.tudelft.ewi.devhub.server.database.entities.Commit;
+import nl.tudelft.ewi.devhub.server.database.entities.RepositoryEntity;
+import nl.tudelft.ewi.devhub.server.database.entities.warnings.LargeCommitWarning;
+import nl.tudelft.ewi.devhub.server.web.models.GitPush;
+import nl.tudelft.ewi.git.models.AbstractDiffModel;
+import nl.tudelft.ewi.git.models.DiffModel;
+import nl.tudelft.ewi.git.web.api.CommitApi;
+import nl.tudelft.ewi.git.web.api.RepositoriesApi;
+import nl.tudelft.ewi.git.web.api.RepositoryApi;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Stream;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Matchers.anyObject;
+import static org.mockito.Matchers.anyString;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@RunWith(MockitoJUnitRunner.class)
+public class LargeCommitWarningGeneratorTest {
+
+    private LargeCommitWarningGenerator generator;
+    private Set<LargeCommitWarning> testEquals;
+
+    private static final String REPOSITORY_NAME = "John Cena";
+    private static final String COMMIT_ID = "1";
+
+    @Mock private RepositoriesApi repositoriesApi;
+    @Mock private Commit commit;
+    @Mock private RepositoryEntity configurable;
+    @Mock private GitPush gitPush;
+    @Mock private List<AbstractDiffModel.DiffFile<AbstractDiffModel.DiffContext<AbstractDiffModel.DiffLine>>> diffs;
+    @Mock private AbstractCommitWarningGenerator abstractCommitWarningGenerator;
+    @Mock private CommitApi commitApi;
+    @Mock private DiffModel diffModel;
+    @Mock private RepositoryApi repositoryApi;
+    @Mock private Collection collection;
+    @Mock private Stream<AbstractDiffModel.DiffFile<AbstractDiffModel.DiffContext<AbstractDiffModel.DiffLine>>> stream;
+    @Mock private Stream<Object> objectStream;
```

Comment:
The `NullPointerException` is likely caused because it is deduced from the `newLineNumber` and `oldLineNumber` fields, in order to save space. Namely: if the line is added, it was not present in the old file, so the `oldLineNumber` is `null`. If the line was removed, it is not present in the new file, and the `newLineNumber` is `null`. If the line was present in both the new and old file, it can only be modified.

The implementation of `isAdded()` and `isRemoved()` uses these fields, and expect at least one of them to be not null. We do not add booleans / an enum value for `added`/`removed`/`modified` as this would require a lot of unnecessary space in the response (and consider it is sent over the wire for every single line in the diff).

As far as creating the diff object

1) Programatically creating a diff file
```java
		DiffLine diffLine = new DiffLine();
		diffLine.setNewLineNumber(1);
		diffLine.setContent("Hiephoi");
		
		DiffContext<DiffLine> diffContext = new DiffContext<>();
		diffContext.setLines(Lists.newArrayList(diffLine));
		DiffFile<DiffContext<DiffLine>> diffFile = new DiffFile<>();
		diffFile.setContexts(Lists.newArrayList(diffContext));
		DiffModel diffModel = new DiffModel();
		diffModel.setDiffs(Lists.newArrayList(diffFile));
```

2)
You can also create an example diff you need for your test, and convert it to JSON (if you are familiar with that). You can then store it as a JSON file under resources, and load it using the ObjectMapper:

```
ObjectMapper objectMapper = new ObjectMapper();
DiffModel diffModel = objectMapper.readValue(
   LargeCommitWarningGeneratorTest.class.getResourceAsStream("/my-diff.json"),
   DiffModel.class
);